streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -17,242 +17,52 @@
     public static final Long WINDOW_CHANGE_LOG_ADDITIONAL_RETENTION_DEFAULT = TimeUnit.MILLISECONDS.convert(1, TimeUnit.DAYS);
 
-    private final ZkClient zkClient;
-    private final int replicationFactor;
     private final long windowChangeLogAdditionalRetention;
 
-    private class ZKStringSerializer implements ZkSerializer {
-
-        /**
-         * @throws AssertionError if the byte String encoding type is not supported
-         */
-        @Override
-        public byte[] serialize(Object data) {
-            try {
-                return ((String) data).getBytes("UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                throw new AssertionError(e);
-            }
-        }
-
-        /**
-         * @throws AssertionError if the byte String encoding type is not supported
-         */
-        @Override
-        public Object deserialize(byte[] bytes) {
-            try {
-                if (bytes == null)
-                    return null;
-                else
-                    return new String(bytes, "UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                throw new AssertionError(e);
-            }
-        }
-    }
-
-    public InternalTopicManager() {
-        this.zkClient = null;
-        this.replicationFactor = 0;
-        this.windowChangeLogAdditionalRetention = WINDOW_CHANGE_LOG_ADDITIONAL_RETENTION_DEFAULT;
-    }
+    private final int replicationFactor;
+    private final StreamsKafkaClient streamsKafkaClient;
 
-    public InternalTopicManager(String zkConnect, final int replicationFactor, long windowChangeLogAdditionalRetention) {
-        this.zkClient = new ZkClient(zkConnect, 30 * 1000, 30 * 1000, new ZKStringSerializer());
+    public InternalTopicManager(final StreamsKafkaClient streamsKafkaClient, final int replicationFactor, final long windowChangeLogAdditionalRetention) {
+        this.streamsKafkaClient = streamsKafkaClient;
         this.replicationFactor = replicationFactor;
         this.windowChangeLogAdditionalRetention = windowChangeLogAdditionalRetention;
     }
 
-    public void makeReady(InternalTopicConfig topic, int numPartitions) {
-        boolean topicNotReady = true;
-
-        while (topicNotReady) {
-            Map<Integer, List<Integer>> topicMetadata = getTopicMetadata(topic.name());
-
-            if (topicMetadata == null) {
-                try {
-                    createTopic(topic, numPartitions, replicationFactor);
-                } catch (ZkNodeExistsException e) {
-                    // ignore and continue
+    /**
+     * If the topic exists do nothing (we don't change partititions or delete the existing topics).
+     * Otherwise create the new topic.
+     * @param topic
+     * @param numPartitions
+     */
+    public void makeReady(final InternalTopicConfig topic, final int numPartitions) {
+
+        if (!streamsKafkaClient.topicExists(topic.name())) {
```

Comment:
Just FYI @hachikuji and I found the original while-loop may be the culprit of the recent unit test hanging issue.